lib/network_interface/worker.ex

```diff
@@ -125,8 +126,37 @@ defmodule Nerves.NetworkInterface.Worker do
     { :ok, %Nerves.NetworkInterface.Worker{port: port} }
   end
 
+  #Returns intersection of lists a and b
+  defp intersect(a, b), do: a -- (a -- b)
+
+  # Returns list of interfaces to be managed by Nerves.NetworkInterface and Nerves.Network modules
+  # By default this is list of ALL network interfaces available in the system. It can be reduced
+  # by specifying a list of interfaces we want to be managed by Nerves.Network sub-system in the 
+  # .../config/config.exs file.
+  defp get_managed_interfaces(available_interfaces) do
+    managed_interfaces = Application.get_env(:nerves_network_interface, :managed_interfaces, [])
+    Logger.debug "#{__MODULE__}: managed_interfaces = #{inspect managed_interfaces}"
+
+    case managed_interfaces do
+      "all" -> available_interfaces
+      nil -> available_interfaces
+      [] -> available_interfaces
+      _ -> managed_interfaces
+    end
+  end
+
+  def handle_call(:all_interfaces, _from, state) do
```

Comment:
for all your protesting, you managed to have nice small functions! ;) looks good!